src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -431,15 +437,34 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
       this.superinterfaces += superinterfaces
     }
 
-    fun addSuperinterface(superinterface: TypeName) = apply {
-      superinterfaces += superinterface
+    fun addSuperinterface(superinterface: TypeName, delegate: CodeBlock = CodeBlock.EMPTY) = apply {
+      if (delegate.isEmpty())
```

Comment:
Curly braces are _always_ required for blocks unless the statement is on the same line as the conditional (which you're only allowed to do if there's no `else`).